sound/soc/sof/ipc.c

```diff
@@ -296,10 +296,7 @@ static int sof_ipc_tx_message_unlocked(struct snd_sof_ipc *ipc, u32 header,
 	ipc_log_header(sdev->dev, "ipc tx", msg->header);
 
 	/* now wait for completion */
-	if (!ret)
-		ret = tx_wait_done(ipc, msg, reply_data);
-
-	return ret;
+	return tx_wait_done(ipc, msg, reply_data);
```

Comment:
there might have been cases when `ret` could be positive, but looks like all those cases are gone by now, or maybe it was never the case and I'm imagining things. Anyway looks correct now.

Reply:
it adds to the confusion that it is semi random how the ret is tested:
`if (ret < 0)` vs `if (!ret)`

`sof_ipc_tx_message_unlocked()` is a nice example as the same `ret` is tested first for `!ret` then for `ret < 0` :
```
ret = snd_sof_dsp_send_msg(sdev, msg);
/* Next reply that we receive will be related to this message */
if (!ret)
	msg->ipc_complete = false;

spin_unlock_irq(&sdev->ipc_lock);

if (ret < 0) {
	dev_err_ratelimited(sdev->dev,
			    "error: ipc tx failed with error %d\n",
			    ret);
	return ret;
}
```